src/cpp/core/file_lock/LinkBasedFileLock.cpp

```diff
@@ -93,10 +94,63 @@ bool isLockFileStale(const FilePath& lockFilePath)
    return LinkBasedFileLock::isLockFileStale(lockFilePath);
 }
 
+bool isLockFileOrphaned(const FilePath& lockFilePath)
+{
+   Error error;
+   
+   // attempt to read pid from lockfile
+   std::string pid;
+   error = core::readStringFromFile(lockFilePath, &pid);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return false;
+   }
+   pid = string_utils::trimWhitespace(pid);
+   
+#ifdef __linux__
+   
+   // on linux, we can check the proc filesystem for an associated
+   // process -- if there is no such directory, then we assume that
+   // this lockfile has been orphaned
+   FilePath procPath("/proc/" + pid);
+   if (!procPath.exists())
+      return true;
```

Comment:
I allowed for fall-through here out of the possibility that we're unable to actually read folders within the `/proc` directory (e.g. if permissions / capabilities on the system were set up in such a way that processes couldn't poke around in there). Is that overly pessimistic?